### PR TITLE
Performance improvement

### PR DESCRIPTION
**Pull request by shirleyqiantingliu**
# Problem / Feature
Remove unnecessary fetch in one action.
Avoid fetching whole jobbatch payload.
Add transactions when finishing Job batch. So that when multiple parallel jobs try to finish the same jobbatch, only one job can do that. (including update job batch, get the results, delete jobbatch)
When job batch is error and the abort mode is abort fast, all not started jobs will be aborted. Then when the job is taken from the queue, there is no need to check job batch's progress status. 
# Changes

### CODE REVIEW

File: src/db/storage/sequelizeStorage/JobBatch.js [hunk 1]
```diff
@@ -38,74 +37,55 @@ const createJobBatch = db => (
...
-
-const countFinishedAndFailedJobs = db => (jobBatchId) => {
+const countFinishedAndFailedJobs = db => (jobBatchId, transaction) => {
   return Promise.props({
     successful: db.models.Job.count({
       where: {
         jobBatchId,
         progressStatus: [PROGRESS_STATUS.FINISHED_SUCCESSFULLY]
-      }
+      },
```
Comment (conckja1) [inline src/db/storage/sequelizeStorage/JobBatch.js:80]: Does the count limit the attributes b default or should we specify `attributes` as well?

Comment (conckja1) [reply src/db/storage/sequelizeStorage/JobBatch.js:80]: same for errors counting  bellow

File: src/db/storage/sequelizeStorage/Job.js [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
+const hasUnfinishedJob = db => async (jobBatchId, transaction) => {
+  const job = await db.models.Job.findOne({
+    attributes: ['id'],
+    where: {
+      jobBatchId,
+      progressStatus: [PROGRESS_STATUS.NOT_STARTED, PROGRESS_STATUS.IN_PROGRESS]
+    },
+    transaction
+  });
+  return !!job ? true : false;
```
Comment (conckja1) [inline src/db/storage/sequelizeStorage/Job.js:31]: `return !!job;` should be enugh

File: src/db/storage/sequelizeStorage/JobBatch.js (hunk 1 above)
Comment (shirleyqiantingliu) [reply src/db/storage/sequelizeStorage/JobBatch.js:80]: sequelize does not support attributes for count()

File: src/db/storage/sequelizeStorage/Job.js (hunk 2 above)
Comment (shirleyqiantingliu) [reply src/db/storage/sequelizeStorage/Job.js:31]: yes. right.

**Pull request merged**